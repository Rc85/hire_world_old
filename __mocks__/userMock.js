export const User = {
    account_type: "Listing",
    allow_messaging: true,
    avatar_url: "/user_files/1/businessman-avatar-icon-01-.jpg",
    display_fullname: true,
    email_notifications: false,
    hide_email: false,
    user_address: "3475 E 27th Ave",
    user_bio: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Assumenda ullam repellat maiores aperiam debitis iure, repudiandae maxime pariatur nam rerum quasi numquam aliquid illo temporibus amet officiis, veritatis ipsum? Illum. Lorem ipsum dolor sit amet consectetur adipisicing elit. Aspernatur similique quae tenetur repellat earum voluptate, reiciendis doloribus. Pariatur atque porro sunt autem quod harum soluta necessitatibus tempora hic. Molestias, harum! Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic molestiae nihil, optio accusantium repellat ex cupiditate commodi similique facere minus maxime explicabo dolorum neque veniam corrupti ipsa debitis. Pariatur, necessitatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus voluptatem, a molestias aperiam obcaecati doloribus harum veritatis suscipit asperiores hic exercitationem molestiae aliquid soluta minima! Laborum placeat saepe dolores! Enim.",
    user_business_name: "RC Web Development",
    user_city: "Vancouver",
    user_city_code: "V5R1P8",
    user_country: "Canada",
    user_education: "BCIT",
    user_email: "rogerchin85@gmail.com",
    user_facebook: "https://www.facebook.com",
    user_firstname: "Roger",
    user_github: "https://github.com/Rc85",
    user_instagram: "https://www.instagram.com",
    user_last_login: "2018-12-10T00:32:16.540Z",
    user_lastname: "Chin",
    user_linkedin: "https://www.linkedin.com",
    user_phone: "604-889-8353",
    user_profile_id: 1,
    user_region: "British Columbia",
    user_setting_id: 1,
    user_title: "Web Developer",
    user_twitter: "https://www.twitter.com",
    user_website: "https://www.rogerchin.me",
    user_worldwide: null,
    username: "roger85"
}