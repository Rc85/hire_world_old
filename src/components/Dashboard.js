import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import '../styles/Dashboard.css';
import Response from './Response';
import EditUser from './EditUser';

class Dashboard extends Component {
    render() {
        let component;

        if (this.props.user) {
            component = <section id='dashboard' className='main-panel'>
                <section className='tab-bar'>
                    <div className='tab-button active'>Edit</div>
                    <div className='tab-button'>My Listings</div>
                    <div className='tab-button'>Offers</div>
                    <div className='tab-button'>Accepted</div>
                    <div className='tab-button'>Settings</div>
                </section>

                <EditUser />
            </section>;
        } else {
            component = <Response code={401} header='Unauthorized Access' message={`You're not logged in`} />;
        }
        
        return(
            <div>{component}</div>
        )
    }
}

const mapStateToProps = state => {
    return {
        user: state.Login.user
    }
}

export default withRouter(connect(mapStateToProps)(Dashboard));