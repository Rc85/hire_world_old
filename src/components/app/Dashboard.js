import React, { Component } from 'react';
import { withRouter, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import SideBar from '../includes/site/SideBar';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCommentAlt, faCog, faSyncAlt, faBriefcase, faLink } from '@fortawesome/pro-solid-svg-icons';
import BottomBar from '../includes/site/BottomBar';
import { connect } from 'react-redux';
import Response from './Response';
import { faFilePlus, faUser } from '@fortawesome/pro-solid-svg-icons';
import fetch from 'axios';
import { PromptOpen, PromptReset } from '../../actions/PromptActions';

class Dashboard extends Component {
    constructor(props) {
        super(props);
    }
    
    componentDidUpdate(prevProps, prevState) {
        if (prevProps.location.key !== this.props.location.key) {
            fetch.post('/api/get/announcements')
            .then(resp => {
                if (resp.data.status === 'success') {
                    let ids = [];
                    let localIds = JSON.parse(localStorage.getItem('dismissed'));

                    for (let a of resp.data.announcements) {
                        if (!localIds) {
                            ids.push(a.announcement_id);
                        } else if (localIds && localIds.indexOf(a.announcement_id) < 0) {
                            ids.push(a.announcement_id);
                        }
                    }

                    this.setState({announcements: resp.data.announcements, announcementIds: ids});
                }
            })
            .catch(err => LogError(err, '/api/get/announcements'));
        }
    }

    componentDidMount() {
        fetch.post('/api/get/announcements')
        .then(resp => {
            if (resp.data.status === 'success') {
                let ids = [];
                let localIds = JSON.parse(localStorage.getItem('dismissed'));

                for (let a of resp.data.announcements) {
                    if (!localIds) {
                        ids.push(a.announcement_id);
                    } else if (localIds && localIds.indexOf(a.announcement_id) < 0) {
                        ids.push(a.announcement_id);
                    }
                }

                this.setState({announcements: resp.data.announcements, announcementIds: ids});
            }
        })
        .catch(err => LogError(err, '/api/get/announcements'));
    }
    
    render() {
        let dashboard;

        let items = [
            {name: 'My Profile', link: '/dashboard/profile', active: /^\/dashboard\/profile/.test(this.props.location.pathname), icon: <FontAwesomeIcon icon={faUser} />},
            {name: 'Conversations', link: '/dashboard/conversations', active: /^\/dashboard\/conversations/.test(this.props.location.pathname), icon: <FontAwesomeIcon icon={faCommentAlt} />, messageCount: parseInt(this.props.user.messages)},
            {name: 'Post a Job', link: '/dashboard/post/job', active: this.props.location.pathname === '/dashboard/post/job', icon: <FontAwesomeIcon icon={faFilePlus} />},
            {name: 'My Jobs', link: '/dashboard/jobs', active: /^\/dashboard\/(jobs|job|posted|applied|saved)/.test(this.props.location.pathname), icon: <FontAwesomeIcon icon={faBriefcase} />, messageCount: parseInt(this.props.user.proposals) + parseInt(this.props.user.job_messages.opened_job_message_count) + parseInt(this.props.user.job_messages.active_job_message_count), items: [
                {name: 'Saved', active: this.props.location.pathname === '/dashboard/saved/jobs', link: '/dashboard/saved/jobs'},
                {name: 'Posted', active: /^\/dashboard\/posted\/(jobs|job)/.test(this.props.location.pathname), link: '/dashboard/posted/jobs'},
                {name: 'Applied', active: this.props.location.pathname === '/dashboard/applied/jobs', link: '/dashboard/applied/jobs'},
                {name: 'Proposals', active: this.props.location.pathname === '/dashboard/jobs/opened', link: '/dashboard/jobs/opened', messageCount: parseInt(this.props.user.proposals) + parseInt(this.props.user.job_messages.opened_job_message_count)},
                {name: 'Active', active: this.props.location.pathname === '/dashboard/jobs/active', link: '/dashboard/jobs/active', messageCount: parseInt(this.props.user.job_messages.active_job_message_count)},
                {name: 'Completed', active: this.props.location.pathname === '/dashboard/jobs/complete', link: '/dashboard/jobs/complete'},
                {name: 'Abandoned', active: this.props.location.pathname === '/dashboard/jobs/abandoned', link: '/dashboard/jobs/abandoned'}
            ]},
            {name: 'Link Work', link: '/dashboard/link_work', active: this.props.location.pathname === '/dashboard/link_work', icon: <FontAwesomeIcon icon={faLink} />},
            {name: 'Subscription', link: '/dashboard/subscription', active: this.props.location.pathname === '/dashboard/subscription', icon: <FontAwesomeIcon icon={faSyncAlt} />},
            {name: 'Settings', link: '/dashboard/settings/account', active: /^\/dashboard\/settings/.test(this.props.location.pathname), icon: <FontAwesomeIcon icon={faCog} />, items: [
                {name: 'Account', active: this.props.location.pathname === '/dashboard/settings/account', link: '/dashboard/settings/account'},
                {name: 'Payment', active: this.props.location.pathname === '/dashboard/settings/payment', link: '/dashboard/settings/payment'},
                {name: 'Link Work', active: this.props.location.pathname === '/dashboard/settings/link_work', link: '/dashboard/settings/link_work'}
            ]}
        ];

        if (this.props.user.status === 'activation required') {
            dashboard = <Response code={403} header={'Forbidden'} message='You need to activate your account'><div><NavLink to='/resend'>Resend Confirmation Email</NavLink></div></Response>;
        } else {
            dashboard = <React.Fragment>
                {this.props.location.pathname.match(/^\/dashboard\/edit$/) ? <div id='main-panel-bg'></div> : ''}      
                {this.props.children}
            </React.Fragment>;
        }

        return(
            <section id='dashboard'>
                {!this.props.config.mobile ? <SideBar user={this.props.user} items={items} /> : <BottomBar user={this.props.user} items={items} />}

                <div id='dashboard-main'>
                    <div className={`dashboard-main-overlay ${this.props.menu.id === 'browse-menu' ? 'show' : ''}`} />
                    {dashboard}
                </div>
            </section>
        );
    }
}

Dashboard.propTypes = {
    user: PropTypes.object.isRequired,
}

const mapStateToProps = state => {
    return {
        config: state.Config,
        user: state.Login,
        prompt: state.Prompt,
        menu: state.Menu
    }
}

export default withRouter(connect(mapStateToProps)(Dashboard));