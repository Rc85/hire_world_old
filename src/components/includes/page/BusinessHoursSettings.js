import React, { Component } from 'react';
import PropTypes from 'prop-types';
import fetch from 'axios';
import SubmitButton from '../../utils/SubmitButton';
import { connect } from 'react-redux';
import { Alert } from '../../../actions/AlertActions';
import { LogError } from '../../utils/LogError';
import SlideToggle from '../../utils/SlideToggle';
import { UpdateUser } from '../../../actions/LoginActions';
import TitledContainer from '../../utils/TitledContainer';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBusinessTime } from '@fortawesome/free-solid-svg-icons';
import InputGroup from '../../utils/InputGroup';
import { IsTyping } from '../../../actions/ConfigActions';

class BusinessHoursSettings extends Component {
    constructor(props) {
        super(props);
        
        this.state = {
            status: '',
            monStartTime: '',
            tueStartTime: '',
            wedStartTime: '',
            thuStartTime: '',
            friStartTime: '',
            satStartTime: '',
            sunStartTime: '',
            monEndTime: '',
            tueEndTime: '',
            wedEndTime: '',
            thuEndTime: '',
            friEndTime: '',
            satEndTime: '',
            sunEndTime: ''
        }
    }

    componentDidMount() {
        fetch.post('/api/get/business_hours', {id: this.props.id})
        .then(resp => {
            if (resp.data.status === 'success') {
                this.initialState = this.splitHours(resp.data.hours);

                this.setState(this.initialState);
            }
        })
        .catch(err => LogError(err, '/api/get/business_hours'));
    }

    splitHours(days) {
        let monStart, monEnd, tueStart, tueEnd, wedStart, wedEnd, thuStart, thuEnd, friStart, friEnd, satStart, satEnd, sunStart, sunEnd;

        for (let day in days) {
            let hours = days[day].split(' - ');

            if (hours.length === 2) {
                if (day === 'monday') {
                    monStart = hours[0];
                    monEnd = hours[1];
                } else if (day === 'tuesday') {
                    tueStart = hours[0];
                    tueEnd = hours[1];
                } else if (day === 'wednesday') {
                    wedStart = hours[0];
                    wedEnd = hours[1];
                } else if (day === 'thursday') {
                    thuStart = hours[0];
                    thuEnd = hours[1];
                } else if (day === 'friday') {
                    friStart = hours[0];
                    friEnd = hours[1];
                } else if (day === 'saturday') {
                    satStart = hours[0];
                    satEnd = hours[1];
                } else if (day === 'sunday') {
                    sunStart = hours[0];
                    sunEnd = hours[1];
                }
            }
        }

        let initialState = {
            status: '',
            monStartTime: monStart,    
            tueStartTime: tueStart,
            wedStartTime: wedStart,
            thuStartTime: thuStart,
            friStartTime: friStart,
            satStartTime: satStart,
            sunStartTime: sunStart,
            monEndTime: monEnd,
            tueEndTime: tueEnd,
            wedEndTime: wedEnd,
            thuEndTime: thuEnd,
            friEndTime: friEnd,
            satEndTime: satEnd,
            sunEndTime: sunEnd
        }

        return initialState;
    }
    
    save(state) {
        this.setState({status: 'Loading'});

        let days = {}

        if (state.monStartTime && state.monEndTime) {
            days['mon'] = state.monStartTime + ' - ' + state.monEndTime;
        } else {
            days['mon'] = 'Closed';
        }

        if (state.tueStartTime && state.tueEndTime) {
            days['tue'] = state.tueStartTime + ' - ' + state.tueEndTime;
        } else {
            days['tue'] = 'Closed';
        }

        if (state.wedStartTime && state.wedEndTime) {
            days['wed'] = state.wedStartTime + ' - ' + state.wedEndTime;
        } else {
            days['wed'] = 'Closed';
        }

        if (state.thuStartTime && state.thuEndTime) {
            days['thu'] = state.thuStartTime + ' - ' + state.thuEndTime;
        } else {
            days['thu'] = 'Closed';
        }

        if (state.friStartTime && state.friEndTime) {
            days['fri'] = state.friStartTime + ' - ' + state.friEndTime;
        } else {
            days['fri'] = 'Closed';
        }

        if (state.satStartTime && state.satEndTime) {
            days['sat'] = state.satStartTime + ' - ' + state.satEndTime;
        } else {
            days['sat'] = 'Closed';
        }

        if (state.sunStartTime && state.sunEndTime) {
            days['sun'] = state.sunStartTime + ' - ' + state.sunEndTime;
        } else {
            days['sun'] = 'Closed';
        }

        fetch.post('/api/user/business_hours/save', {days: days, id: this.props.id})
        .then(resp => {
            if (resp.data.status === 'success') {
                this.initialState = {...this.state};
                this.initialState.status = '';
            }

            this.setState(this.initialState);
            this.props.dispatch(Alert(resp.data.status, resp.data.statusMessage));
        })
        .catch(err => LogError(err, '/api/user/business_hours/save'));
    }

    toggle() {
        this.setState({status: 'Loading'});

        let data = {...this.props.user.user};
        data['display_business_hours'] = !data['display_business_hours'];

        fetch.post(`/api/user/settings/change`, data)
        .then(resp => {
            this.setState({status: ''});

            if (resp.data.status === 'success') {
                this.props.dispatch(UpdateUser(resp.data.user));
            } else {
                this.props.dispatch(Alert(resp.data.status, resp.data.statusMessage));
            }
        })
        .catch(err => LogError(err, '/api/user/settings/change'));
    }
    
    render() {
        return (
            <section id='business-hours-settings' className='mb-3'>
                <div className='d-flex-between-center'>
                    <div className='mr-1'><h5>Business Hours:</h5></div>

                    <SlideToggle status={this.props.user.user.display_business_hours} onClick={() => this.toggle()} />
                </div>

                <span>If one or both fields are blank, it will indicate 'Closed' for that day.</span>

                <hr/>

                <div id='hours-settings'>
                    <form onSubmit={(e) => {
                        e.preventDefault();
                        this.save(this.state);
                    }}>
                        <HourSetters day='Monday' startTime={(val) => this.setState({monStartTime: val})} endTime={(val) => this.setState({monEndTime: val})} startValue={this.state.monStartTime} endValue={this.state.monEndTime} dispatch={this.props.dispatch} />
                        <HourSetters day='Tuesday' startTime={(val) => this.setState({tueStartTime: val})} endTime={(val) => this.setState({tueEndTime: val})} startValue={this.state.tueStartTime} endValue={this.state.tueEndTime} dispatch={this.props.dispatch} />
                        <HourSetters day='Wednesday' startTime={(val) => this.setState({wedStartTime: val})} endTime={(val) => this.setState({wedEndTime: val})} startValue={this.state.wedStartTime} endValue={this.state.wedEndTime} dispatch={this.props.dispatch} />
                        <HourSetters day='Thursday' startTime={(val) => this.setState({thuStartTime: val})} endTime={(val) => this.setState({thuEndTime: val})} startValue={this.state.thuStartTime} endValue={this.state.thuEndTime} dispatch={this.props.dispatch} />
                        <HourSetters day='Friday' startTime={(val) => this.setState({friStartTime: val})} endTime={(val) => this.setState({friEndTime: val})} startValue={this.state.friStartTime} endValue={this.state.friEndTime} dispatch={this.props.dispatch} />
                        <HourSetters day='Saturday' startTime={(val) => this.setState({satStartTime: val})} endTime={(val) => this.setState({satEndTime: val})} startValue={this.state.satStartTime} endValue={this.state.satEndTime} dispatch={this.props.dispatch} />
                        <HourSetters day='Sunday' startTime={(val) => this.setState({sunStartTime: val})} endTime={(val) => this.setState({sunEndTime: val})} startValue={this.state.sunStartTime} endValue={this.state.sunEndTime} dispatch={this.props.dispatch} />
    
                        <div className='text-right'><SubmitButton loading={this.state.status === 'Loading'} type='submit' value='Save' disabled={JSON.stringify(this.state) == JSON.stringify(this.initialState)} /></div>
                    </form>
                </div>
            </section>
        );
    }
}

const HourSetters = props => {
    return(
        <div id={props.day} className='hour-container'>                      
            <InputGroup className='hour-container' label={props.day}>
                <div className='start-time'><input type='text' onChange={(e) => props.startTime(e.target.value)} maxLength='15' defaultValue={props.startValue} onFocus={() => props.dispatch(IsTyping(true))} onBlur={() => props.dispatch(IsTyping(false))} /></div>
                <div className='hour-separator'>to</div>
                <div className='end-time'><input type='text' onChange={(e) => props.endTime(e.target.value)} maxLength='15' defaultValue={props.endValue} onFocus={() => props.dispatch(IsTyping(true))} onBlur={() => props.dispatch(IsTyping(false))} /></div>
            </InputGroup>
        </div>
    )
}

BusinessHoursSettings.propTypes = {
    user: PropTypes.object
}

const mapStateToProps = state => {
    return {
        user: state.Login
    }
}

export default connect(mapStateToProps)(BusinessHoursSettings);