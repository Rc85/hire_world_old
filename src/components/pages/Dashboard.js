import React from 'react';
import { withRouter } from 'react-router-dom';
import Response from './Response';
import TabBar from '../utils/TabBar';
import Loading from '../utils/Loading';
import PropTypes from 'prop-types';

const Dashboard = props => {
    let success = /success$/;
    let fail = /fail$/;

    if (success.test(props.user.status) || props.user.user) {
        console.log(props.user);

        return(
            <section id='dashboard' className='main-panel w-100'>
                <TabBar items={[
                    {name: 'Edit', active: props.location.pathname === '/dashboard/edit', link: '/dashboard/edit'},
                    {name: 'List', active: props.location.pathname === '/dashboard/list', link: '/dashboard/list'},
                    {name: 'Inquiries', active: /^\/dashboard\/message(s)?\/Inquire.*/.test(props.location.pathname), link: '/dashboard/messages/Inquire'},
                    {name: 'Active Jobs', active: /^\/dashboard\/message(s)?\/Active.*/.test(props.location.pathname), link: '/dashboard/messages/Active'},
                    {name: 'Completed', active: /^\/dashboard\/message(s)?\/Complete.*/.test(props.location.pathname), link: '/dashboard/messages/Complete'},
                    {name: 'Abandoned', active: /^\/dashboard\/message(s)?\/Abandoned.*/.test(props.location.pathname), link: '/dashboard/messages/Abandoned'},
                    {name: 'Settings', active: props.location.pathname === '/dashboard/settings', link: '/dashboard/settings'},
                ]} />

                {props.children}
            </section>
        )
    } else if (fail.test(props.user.status)) {
        return(
            <Response code={401} header='Unauthorized Access' message={`You're not logged in`} />
        )
    } else if (!props.user.status) {
        return(
            <Loading size='7x' />
        )
    } else if (!props.user.user) {
        return(
            <Response code={500} header='Internal Server Error' message={'An unknown error has occurred. Please contact the administrator.'} />
        )
    } else {
        return(
            <div></div>
        )
    }
}

Dashboard.propTypes = {
    user: PropTypes.object.isRequired
}

export default withRouter(Dashboard);