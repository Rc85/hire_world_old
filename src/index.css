@font-face {
    font-family: 'Montserrat';
    src: url('./fonts/Montserrat-Regular.ttf');
}

@font-face {
    font-family: 'Montserrat Bold';
    src: url('./fonts/Montserrat-Bold.ttf');
}

html {
    min-height: 100vh;
}

body {
    margin: 0;
    padding: 0;
    font-family: 'Montserrat';
    background: #C4C4C4;
    color: #F9F6F4;
    min-height: 100vh;
    display: flex;
    overflow-x: hidden;
}

#root {
    width: 100%;
    min-height: 100%;
}

a {
    font-weight: bold;
    color: #F9F6F4;
}

a:hover {
    color: #F9F6F4;
    text-decoration: underline;
}

h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat Bold';
}

hr {
    background: #2988BC;
}

/* Widths */

.w-5 {
    width: 5%;
}

.w-10 {
    width: 10%;
}

.w-20 {
    width: 20%;
}

.w-30 {
    width: 30%;
}

.w-33 {
    width: 33%;
}

.w-35 {
    width: 35%;
}

.w-40 {
    width: 40%;
}

.w-45 {
    width: 45%;
}

.w-55 {
    width: 55%;
}

.w-60 {
    width: 60%;
}
