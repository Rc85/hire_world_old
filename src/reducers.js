import { combineReducers } from 'redux';
import { ToggleMenu } from './reducers/Togglers';
import { Login } from './reducers/Auths';
import { Sectors } from './reducers/Sectors';
import { Confirmation } from './reducers/Confirmation';

export const reducers = combineReducers({
    ToggleMenu, Login, Sectors, Confirmation
});