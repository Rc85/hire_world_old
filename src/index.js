import 'babel-polyfill';
import 'normalize.css';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import './styles/vendor/bootstrap.css';
import './styles/vendor/bootstrap-grid.css';
import './styles/vendor/bootstrap-reboot.css';
import './styles/index.css';
import App from './App';
import { Provider } from 'react-redux';
import { reducers } from './reducers';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

const store = createStore(reducers, applyMiddleware(thunk));
store.subscribe(() => { console.log(store.getState()) });

ReactDOM.render(
    <Provider store={store}>
        <Router basename='/'>
            <App />
        </Router>
    </Provider>,
    document.getElementById('root')
);