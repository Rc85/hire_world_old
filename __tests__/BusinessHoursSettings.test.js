import React from 'react';
import Enzyme, { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router';
import Dashboard from '../src/components/pages/Dashboard';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { User } from '../__mocks__/userMock';
import moxios from 'moxios';
import { createStore, applyMiddleware } from 'redux';
import { reducers } from '../src/reducers';
import BusinessHoursSettings from '../src/components/includes/page/BusinessHoursSettings';
import axios from 'axios';

const store = createStore(reducers, applyMiddleware(thunk));
const user = {user: User, status: 'get session success'};

Enzyme.configure({adapter: new Adapter()});
jest.setTimeout(30000);

describe('Business hours setting component', () => {
    test('Snapshot', async() => {
        const wrapper = render(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{pathname: '/dashboard/edit', key: 'response'}]}>
                    <Dashboard user={user}><BusinessHoursSettings /></Dashboard>
                </MemoryRouter>
            </Provider>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Renders', () => {
        const wrapper = render(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{pathname: '/dashboard/edit', key: 'response'}]}>
                    <Dashboard user={user}><BusinessHoursSettings /></Dashboard>
                </MemoryRouter>
            </Provider>
        );

        let button = wrapper.find('button');

        expect(button[0].children[0].attribs).toHaveProperty('data-icon');
        expect(button[0].children[0].attribs['data-icon']).toEqual('caret-down');
    });

    test('Clicking button should show settings', () => {
        const wrapper = shallow(<BusinessHoursSettings store={store} />).dive();
        
        wrapper.find('button').simulate('click');

        expect(wrapper.state()).toHaveProperty('showSettings', true);
        expect(wrapper.render()[0].children[0].next.children[0].attribs.id).toEqual('Monday');
    });

    describe('Handling data', () => {
        beforeEach(() => {
            moxios.install();
        });

        afterEach(() => {
            moxios.uninstall();
        });

        const wrapper = shallow(<BusinessHoursSettings store={store} />).dive();

        test('Fetching data when component mounts', (done) => {
            moxios.stubRequest('/api/get/business_hours', {
                status: 200,
                response: {
                    status: 'success',
                    hours: {
                        monday: '8:00 AM - 8:00 PM',
                        tuesday: '',
                        wednesday: '',
                        thursday: '',
                        friday: '',
                        saturday: '',
                        sunday: ''
                    }
                }
            });

            axios.get('/api/get/business_hours')
            .then(resp => {
                if (resp.data.status === 'success') {
                    let initialState = wrapper.instance().splitHours(resp.data.hours);

                    wrapper.setState(initialState);

                    expect(resp.data.hours['monday']).toEqual('8:00 AM - 8:00 PM');
                    expect(wrapper.state('monStartTime')).toEqual('8:00 AM');
                }
                
                done();
            });
        });
    });
});