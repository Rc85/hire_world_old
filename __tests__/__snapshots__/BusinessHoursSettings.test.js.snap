// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business hours setting component Snapshot 1`] = `
<section
  class="main-panel w-100"
  id="dashboard"
>
  <div
    class="tab-bar"
  >
    <a
      aria-current="page"
      class="active"
      href="/dashboard/edit"
    >
      <div
        class="tab-button active"
      >
        Profile
      </div>
    </a>
    <a
      href="/dashboard/saved_listings"
    >
      <div
        class="tab-button"
      >
        Saved Listings
      </div>
    </a>
    <a
      href="/dashboard/settings"
    >
      <div
        class="tab-button"
      >
        Account Settings
      </div>
    </a>
  </div>
  <div
    class="mb-3"
    id="business-hours-settings"
  >
    <div
      class="d-flex-between-center mb-3"
    >
      <div
        class="d-flex-between-center"
      >
        <div
          class="mr-1"
        >
          Business Hours:
        </div>
        <div
          id="business-hours-tip"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-question-circle fa-w-16 "
            data-icon="question-circle"
            data-prefix="far"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <button
        class="btn btn-info btn-sm"
        id="toggle-hours-setting-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-caret-down fa-w-10 "
          data-icon="caret-down"
          data-prefix="fas"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</section>
`;
